### PR TITLE
[WIP] Add UI for IDM cert pinning

### PR DESCRIPTION
**Pull request by ciaranRoche**
## Motivation
JIRA: https://issues.jboss.org/browse/AEROGEAR-1988

## Description
A panel is added to the current plugin under the service list options which allows the user to bring up a new dialog box. The dialog box allows a user to see a list of pinned certs, along with options to be able to delete, edit and add to the list.

## Progress
- [x] Implemented basic UI
- [ ] Integrate UI with SDK piece

### CODE REVIEW

**Comment by ciaranRoche** [comment]
@secondsun @aidenkeating Would ye mind having a look at what is done so far.

File: src/main/java/org/aerogear/plugin/intellij/mobile/ui/certconfiguration/CertPanel.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
+package org.aerogear.plugin.intellij.mobile.ui.certconfiguration;
+
+import javax.swing.JButton;
+import javax.swing.JLabel;
+
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
```
Comment (secondsun) [inline src/main/java/org/aerogear/plugin/intellij/mobile/ui/certconfiguration/CertPanel.java]: Remove nag headers.

File: src/main/java/org/aerogear/plugin/intellij/mobile/ui/certconfiguration/PinnedCertDialog.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
+package org.aerogear.plugin.intellij.mobile.ui.certconfiguration;
+
+
+import com.intellij.openapi.ui.DialogWrapper;
+import com.intellij.ui.JBColor;
+import org.aerogear.plugin.intellij.mobile.ui.certconfiguration.pinnedcerts.PinnedCertPanel;
+import org.jetbrains.annotations.Nullable;
+
+import javax.swing.*;
```
Comment (secondsun) [inline src/main/java/org/aerogear/plugin/intellij/mobile/ui/certconfiguration/PinnedCertDialog.java]: Don't use wildcard imports.

File: src/main/java/org/aerogear/plugin/intellij/mobile/ui/certconfiguration/pinnedcerts/PinnedCertPanel.java [hunk 3]
```diff
@@ -0,0 +1,230 @@
+package org.aerogear.plugin.intellij.mobile.ui.certconfiguration.pinnedcerts;
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
```
Comment (secondsun) [inline src/main/java/org/aerogear/plugin/intellij/mobile/ui/certconfiguration/pinnedcerts/PinnedCertPanel.java]: Nag headers

File: src/main/java/org/aerogear/plugin/intellij/mobile/ui/certconfiguration/pinnedcerts/PinnedCertPanel.java [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+
+import com.intellij.openapi.components.ServiceManager;
+import org.aerogear.plugin.intellij.mobile.services.MobileNotificationsService;
+
+import javax.swing.*;
```
Comment (secondsun) [inline src/main/java/org/aerogear/plugin/intellij/mobile/ui/certconfiguration/pinnedcerts/PinnedCertPanel.java]: Don't use wildcard imports.

File: src/main/java/org/aerogear/plugin/intellij/mobile/ui/certconfiguration/pinnedcerts/PinnedCertPanel.java [hunk 5]
```diff
@@ -0,0 +1,230 @@
...
+        jLabel4 = new javax.swing.JLabel();
+        jScrollPane1 = new javax.swing.JScrollPane();
+        tableCerts = new javax.swing.JTable();
+        buttonDelete = new javax.swing.JButton();
+        buttonEdit = new javax.swing.JButton();
+        buttonSave = new javax.swing.JButton();
+        jLabel2 = new javax.swing.JLabel();
+        tfBuildType = new javax.swing.JTextField();
+        jLabel6 = new javax.swing.JLabel();
+        jLabel3 = new javax.swing.JLabel();
```
Comment (secondsun) [inline src/main/java/org/aerogear/plugin/intellij/mobile/ui/certconfiguration/pinnedcerts/PinnedCertPanel.java]: Can you make sure all the variables have a descriptive name?

**Comment by vchepeli** [comment]
@ciaranRoche any updates here?

**Comment by aidenkeating** [comment]
@vchepeli This has been put on hold while we look at other approaches for certificate pinning

**Comment by vchepeli** [comment]
closing this one. @ciaranRoche , @aidenkeating feel free to open new one when you have changes ready ;)

**Pull request closed without merging**